docs/spire.md

```diff
@@ -0,0 +1,273 @@
+ │           │     │           Tekton TaskRun              │     │
+ │           │     └───────────────────────────────────────┘     │
+ │  Configure│                                          ▲        │ Attest
+ │  Pod &    │                                          │        │   +
+ │  check    │                                          │        │ Request
+ │  ready    │     ┌───────────┐                        │        │ SVIDs
+ │           └────►│  TaskRun  ├────────────────────────┘        │
+ │                 │  Pod      │                                 │
+ │                 └───────────┘     TaskRun Entrypointer        │
+ │                   ▲               Sign Result and update      │
+ │ Get               │ Get SVID      TaskRun status with         │
+ │ SPIRE             │               signature + cert            │
+ │ server            │                                           │
+ │ Credentials       │                                           ▼
+┌┴───────────────────┴─────────────────────────────────────────────────────┐
+│                                                                          │
+│   SPIRE Agent    ( Runs as   )                                           │
+│   + CSI Driver   ( Daemonset )                                           │
+│                                                                          │
+└──────────────────────────────────────────────────────────────────────────┘
+```
+
+Initial Setup:
+1. As part of the SPIRE deployment, the SPIRE server attests the agents running on each node in the cluster.
+1. The Tekton Controller is configured to have workload identity entry creation permissions to the SPIRE server.
+1. As part of the Tekton Controller operations, the Tekton Controller will retrieve an identity that it can use to talk to the SPIRE server to register TaskRun workloads.
+
+When a TaskRun is created:
+1. The Tekton Controller creates a TaskRun pod and its associated resources
+1. When the TaskRun pod is ready, the Tekton Controller registers an identity with the information of the pod to the SPIRE server. This will tell the SPIRE server the identity of the TaskRun to use as well as how to attest the workload/pod.
+1. After the TaskRun steps complete, as part of the entrypointer code, it requests an SVID from SPIFFE workload API (via the SPIRE agent socket)
+1. The SPIRE agent will attest the workload and request an SVID.
+1. The entrypointer receives an x509 SVID, containing the x509 certificate and associated private key. 
+1. The entrypointer signs the results of the TaskRun and emits the signatures and x509 certificate to the TaskRun results for later verification.
+
+## Enabling TaskRun result attestations
+
+To enable TaskRun attestations:
+1. Make sure `enforce-nonfalsifiability` is set to `"spire"` in the `feature-flags` configmap, see [`install.md`](./install.md#customizing-the-pipelines-controller-behavior) for details
```

Comment:
i dont see any updates to install.md in this PR, could it be updated to mention the new config option as well?